scenario_player/utils/token.py

```diff
@@ -399,19 +405,20 @@ def update_allowance(self) -> Union[Tuple[str, int], None]:
         required_allowance = self.config.settings.services.udc.token.balance_per_node * node_count
 
         log.debug(
-            "Checking necessity of deposit request",
-            required_balance=required_allowance,
-            actual_balance=udt_allowance,
+            "Checking UDTC allowance",
+            required_allowance=required_allowance,
+            required_per_node=self.config.settings.services.udc.token.balance_per_node,
+            node_count=node_count,
+            actual_allowance=udt_allowance,
         )
 
         if not udt_allowance < required_allowance:
-            log.debug("allowance update call not required - sufficient allowance")
+            log.debug("UDTC allowance sufficient")
             return
 
-        log.debug("allowance update call required - insufficient allowance")
-        allow_amount = required_allowance - udt_allowance
+        log.debug("UDTC allowance insufficient, updating")
         params = {
-            "amount": allow_amount,
+            "amount": required_allowance,
```

Comment:
isn't this a regression?

Reply:
No, that's the bugfix

Reply:
Sorry, let me rephrase the question. What was wrong in the previous version?